src/representations/representation_manager_sorted_coulomb.hh

```diff
@@ -0,0 +1,467 @@
+  }
+
+  /* ---------------------------------------------------------------------- */
+
+  template<class Mngr, Option SortAlgo>
+  void RepresentationManagerSortedCoulomb<Mngr, SortAlgo>::compute() {
+    // initialise the sorted coulomb_matrices in linear storage
+    this->coulomb_matrices.resize_to_zero();
+    this->coulomb_matrices.set_nb_row(this->get_n_feature());
+
+
+    // initialize the sorted linear coulomb matrix
+    Eigen::MatrixXd lin_sorted_coulomb_mat(this->size*(this->size+1)/2, 1);
+    //Eigen::MatrixXd coulomb_mat(this->size,this->size);
+
+    // loop over the centers
+    for (auto center : this->structure_manager) {
+      // re-use the temporary coulomb mat in linear storage
+      // need to be zeroed because old data might not be overwritten
+      lin_sorted_coulomb_mat =
+              Eigen::MatrixXd::Zero(this->size*(this->size+1)/2, 1);
+
+      // Nneighbour counts the central atom and the neighbours
+      size_t Nneighbour{center.size()+1};
+
+      // the local distance matrix. Ones to avoid overflow in the div.
+      Eigen::MatrixXd distance_mat =
+              Eigen::MatrixXd::Ones(Nneighbour, Nneighbour);
+      // the matrix with the prefactor on the .
+      Eigen::MatrixXd type_factor_mat =
+              Eigen::MatrixXd::Zero(Nneighbour, Nneighbour);
+
+      // the local coulomb matrix
+      Eigen::MatrixXd coulomb_mat =
+              Eigen::MatrixXd::Ones(Nneighbour, Nneighbour);
+
+
+      this->get_distance_matrix(center, distance_mat, type_factor_mat);
+
+      // Compute Coulomb Mat element wise.
+      coulomb_mat = type_factor_mat.array() / distance_mat.array();
+
+      using Sorter = SortCoulomMatrix<SortAlgo>;
+      switch (SortAlgo) {
+        case Option::CMSortDistance: {
+          auto sort_order{
+                Sorter::get_coulom_matrix_sorting_order(distance_mat)};
+          // inject the coulomb matrix into the sorted linear storage
+          this->sort_and_linearize_coulomb_matrix(coulomb_mat,
+                lin_sorted_coulomb_mat, sort_order);
+          break;
+        }
+        case Option::CMSortRowNorm: {
+          auto sort_order{
+                Sorter::get_coulom_matrix_sorting_order(coulomb_mat)};
+          // inject the coulomb matrix into the sorted linear storage
+          this->sort_and_linearize_coulomb_matrix(coulomb_mat,
+                lin_sorted_coulomb_mat, sort_order);
+          break;
+        }
+      }
+
+      this->coulomb_matrices.push_back(lin_sorted_coulomb_mat);
+    }
+  }
+
+
+  template<class Mngr, Option SortAlgo>
+  void RepresentationManagerSortedCoulomb<Mngr, SortAlgo>::get_distance_matrix(
+    RepresentationManagerSortedCoulomb<Mngr, SortAlgo>::ClusterRef_t<1>& center,
+    Eigen::Ref<Eigen::MatrixXd> distance_mat,
+    Eigen::Ref<Eigen::MatrixXd> type_factor_mat) {
+
+    // the coulomb mat first row and col corresponds
+    // to central atom to neighbours
+    int Zk{center.get_atom_type()};
```

Comment:
I state it again:
We are not programming for ourselves. We have decided on a coding convention. Anything that hinders readability should be rectified. Using `ii` instead of `i` is one of those cases in my opinion. As I said before: the first thing I did was searching for an outer loop. Which did not exist. Which makes the question not something of preference. A lot of people will read it. I am the first one and also the first one to stumble over this.

I agree, the coding convention does not  compel it, **but** the example loops we have in there use an `i` - because it is understandable, everybody knows how to read it and it does not lead to ambiguities.

Reply:
I thought I could be spared from spelling out why I would rather have a repeated single letter iteration variable but let's do it.
First, I read it better, i and j for instance are very small and don't stand out to me. Second, if I want to search or/and replace for this variable i exist in many other variable while ii would most likely only exist for this particular variable. Last, I would rather not address your personal preconceptions by adding a new bullet point in the coding convention on the fact that ii should be part of an inner loop lead by a loop over i.
Are you convinced ?

Reply:
No, sorry. I find it misleading as I explained.
And you give that argument: "I" read it better/don't stand out to "me"/if "I" want to search.

I am ok with developing in that style - you did that, and that is fine. Now that the code is being reviewed and used it is mostly being read, it is confusing.

Maybe someone else should comment on that, too.
I do not feel this is a matter of personal preference/preconception.